crates/tx/src/types.rs

```diff
@@ -1039,6 +1084,27 @@ impl Tx {
         }
     }
 
+    /// Add new default commitments to the transaction. Returns false if the
+    /// commitment is already contained in the set
+    #[cfg(any(test, feature = "testing"))]
+    pub fn push_default_inner_tx(&mut self) -> bool {
+        self.header.batch.insert(TxCommitments::default())
+    }
+
+    /// Add a new inner tx to the transaction. Returns `false` if the
+    /// commitments already existed in the collection. This function expects a
+    /// transaction carrying a single inner tx as input
+    pub fn add_inner_tx(&mut self, other: Tx, cmt: TxCommitments) -> bool {
```

Comment:
If this function is limited to a single inner tx, then I guess the pair `(other, cmt)` is conceptually a `BatchedTx`...

Reply:
Ah you are right! Still, I think it would just be better to rework this function completely in the near future